### PR TITLE
add functionality for highlighting code block, implement copy button …

### PR DESCRIPTION
**Pull request by OleksandrDragun**
…for copying code block

### PR Checklist

* [x] Have you verified that the PR is pointing to the correct target branch? (`develop` for features/bugfixes, `master` for release/hotfix)?
* [x] Have you followed the guidelines in our [Dev Guide](../#readme)
* [x] Have you run prettier (`npm run format`)?
* [x] Have you checked that the build output looks according to design (screen, mobile, desktop) in Figma (`npm run serve`)?

## Visuals

It looks like this
image
image

### CODE REVIEW

File: src/containers/BlogPostPage/constants.tsx [hunk 1]
```diff
@@ -2,10 +2,19 @@
 import React from 'react';
 import { GatsbyImage } from 'gatsby-plugin-image';
 import { BLOCKS, INLINES, MARKS } from '@contentful/rich-text-types';
+import { CopyToClipboardButton } from '@app/components/CopyToClipboardButton';
 
 export const OPTIONS = {
   renderMark: {
     [MARKS.BOLD]: text => <b className="blog-post-page__bold-text">{text}</b>,
+    [MARKS.CODE]: text => {
+      return (
```
Comment (ViktorSoroka07) [inline src/containers/BlogPostPage/constants.tsx]:
```suggestion
    [MARKS.CODE]: text => (
```

File: src/components/CopyToClipboardButton/CopyToClipboardButton.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }, 5000);
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      clearTimeout(id);
+    };
+  }, [isCopied]);
+
+  const onClick = () => {
+    navigator.clipboard.writeText(text).then(() => setCopied(true));
```
Comment (ViktorSoroka07) [inline src/components/CopyToClipboardButton/CopyToClipboardButton.tsx]: I think some error handling should be added. Navigator might not always be present. See the officical documentation

https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText

File: src/containers/BlogPostPage/constants.tsx (hunk 1 above)
Comment (OleksandrDragun) [reply src/containers/BlogPostPage/constants.tsx]: Done.

File: src/components/CopyToClipboardButton/CopyToClipboardButton.tsx (hunk 2 above)
Comment (OleksandrDragun) [reply src/components/CopyToClipboardButton/CopyToClipboardButton.tsx]: Done. Thanks.

**Review by ViktorSoroka07 — DISMISSED** [review]

**Review by AmsterGet — DISMISSED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**